x/retry/retry.go

```diff
@@ -163,7 +163,7 @@ func getTimeLeft(ctx context.Context, max time.Duration) (timeleft time.Duration
 func isRetryable(err error) bool {
 	// TODO(#1080) Update Error assertions to be more granular.
 	switch yarpcerrors.ErrorCode(err) {
-	case yarpcerrors.CodeInternal, yarpcerrors.CodeDeadlineExceeded:
+	case yarpcerrors.CodeInternal, yarpcerrors.CodeDeadlineExceeded, yarpcerrors.CodeUnavailable, yarpcerrors.CodeUnknown:
```

Comment:
I don’t think we can retry Internal or Unknown without more information about the procedure, that is, whether it is idempotent. Check out the TChannel doc for Unexpected (which maps to Unknown or Internal).

https://github.com/uber/tchannel/blob/master/docs/protocol.md#error-0xff